# Generated by Django 5.2 on 2025-07-21 10:08

from django.conf import settings
from django.db import migrations, models


class Migration(migrations.Migration):

    dependencies = [
        ('amazon', '0037_amazonproduct_product_owner'),
        ('core', '0007_alter_multitenantcompany_ai_points_and_more'),
        ('properties', '0013_internal_name_unique'),
        ('sales_channels', '0037_alter_remoteproperty_unique_together_and_more'),
        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
    ]

    operations = [
        migrations.AlterConstraint(
            model_name='amazonproducttype',
            name='unique_amazonproducttype_local_instance_sales_channel_not_null',
            constraint=models.UniqueConstraint(condition=models.Q(('local_instance__isnull', False)), fields=('local_instance', 'sales_channel'), name='unique_amazonproducttype_local_instance_sales_channel_not_null', violation_error_message='An Amazon product type with this local rule already exists for this sales channel.'),
        ),
        migrations.AlterConstraint(
            model_name='amazonproducttype',
            name='unique_amazonproducttype_code_sales_channel_not_null',
            constraint=models.UniqueConstraint(condition=models.Q(('product_type_code__isnull', False)), fields=('product_type_code', 'sales_channel'), name='unique_amazonproducttype_code_sales_channel_not_null', violation_error_message='An Amazon product type with this product type code already exists for this sales channel.'),
        ),
    ]
